maitake/src/wait/cell.rs

```diff
@@ -306,45 +399,63 @@ pub(crate) mod test_util {
 #[cfg(all(loom, test))]
 mod loom {
     use super::*;
-    use crate::loom::{future, thread};
-
-    const NUM_NOTIFY: usize = 2;
+    use crate::loom::{future, sync::Arc, thread};
+    use futures::{select_biased, FutureExt};
 
     #[test]
-    fn basic_latch() {
+    fn basic() {
         crate::loom::model(|| {
-            let chan = test_util::Chan::new(NUM_NOTIFY);
-
-            for _ in 0..NUM_NOTIFY {
-                let chan = chan.clone();
-
-                thread::spawn(move || chan.notify());
-            }
-
-            future::block_on(chan.wait());
-        });
-    }
+            let wake = Arc::new(WaitCell::new());
 
-    #[test]
-    fn close() {
-        crate::loom::model(|| {
-            let chan = test_util::Chan::new(NUM_NOTIFY);
+            let waiter = wake.clone();
+            let closer = wake.clone();
 
-            thread::spawn({
-                let chan = chan.clone();
-                move || {
-                    chan.notify();
-                }
+            thread::spawn(move || {
+                info!("waking");
+                waiter.wake();
+                info!("woke'd");
```

Comment:
extremely silly, sorry: there _is_ actually a word that means "woke'd", it's "woken"...but this is funny, so i am fine with leaving it.